app/controllers/api/v1/messages_controller.rb

```diff
@@ -0,0 +1,20 @@
+module Api
+  module V1
+    class MessagesController < ApplicationController
+      before_action :authenticate_user!
+      before_action :set_conversation_messages, only: [:index]
+
+      def index
+        @messages = @conversation.messages
+      end
+
+      private
+
+      def set_conversation_messages
```

Comment:
I would rename this method to just `conversation` and do like this:
```
def conversation
    @conversation ||= current_user.conversations.find(conversation_id)
end

def conversation_id
    params[:conversation_id]
end
```
Assuming overriding the `has_many :conversations` in the user model works.